### PR TITLE
Detail view

### PR DESCRIPTION
**Pull request by JessLovell**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,7 +18,12 @@
...
-        <activity android:name=".CreateSighting"></activity>
+        <activity android:name=".CreateSighting"
+            android:parentActivityName=".MainActivity"/>
+
+        <activity android:name=".ViewSighting"
+            android:parentActivityName=".MainActivity"
+            android:launchMode="singleTop"></activity>
     </application>
 
 </manifest>
```
Comment (evanslaton) [inline app/src/main/AndroidManifest.xml]: Please add an extra empty line at the end

File: app/src/main/java/com/hwang/xsighting/ViewSighting.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    private final String TAG = "SightingDetail";
+    private Sighting sightingToDisplay;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_view_sighting);
+
+
```
Comment (evanslaton) [inline app/src/main/java/com/hwang/xsighting/ViewSighting.java]: Too many empty lines

File: app/src/main/res/layout/activity_view_sighting.xml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        android:layout_marginEnd="8dp"
+        android:layout_marginBottom="16dp"
+        android:text="by user"
+        android:textSize="18sp"
+        app:layout_constraintBottom_toTopOf="@+id/textView6"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/postLocation"
+        app:layout_constraintTop_toBottomOf="@+id/postTtle" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (jeffborda) [inline app/src/main/res/layout/activity_view_sighting.xml]: new line at the end

File: app/src/main/java/com/hwang/xsighting/ViewSighting.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                        TextView date = findViewById(R.id.postDate);
+                        TextView location = findViewById(R.id.postLocation);
+                        TextView user = findViewById(R.id.postUser);
+                        TextView description = findViewById(R.id.postDescription);
+
+                        date.setText("Today");
+                        location.setText(sightingToDisplay.getLocation());
+                        user.setText(sightingToDisplay.getAuthorUsername());
+                        description.setText(sightingToDisplay.getDescription());
+
```
Comment (evanslaton) [inline app/src/main/java/com/hwang/xsighting/ViewSighting.java]: Too many empty lines

File: app/src/main/java/com/hwang/xsighting/ViewSighting.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+                    } else {
+                        Log.d(TAG, "No such document");
+
+                        //TODO: Redirect to the main page with a failure toast
+
```
Comment (evanslaton) [inline app/src/main/java/com/hwang/xsighting/ViewSighting.java]: Too many empty lines

File: app/src/main/java/com/hwang/xsighting/ViewSighting.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                        int duration = Toast.LENGTH_SHORT;
+
+                        Toast toast = Toast.makeText(context, text, duration);
+                        toast.show();
+                        finish();
+                    }
+                } else {
+                    Log.d(TAG, "get failed with ", task.getException());
+                    //TODO: Redirect to the main page with a failure toast
+
```
Comment (evanslaton) [inline app/src/main/java/com/hwang/xsighting/ViewSighting.java]: Empty line should be above the comment

File: app/src/main/java/com/hwang/xsighting/ViewSighting.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                    int duration = Toast.LENGTH_SHORT;
+
+                    Toast toast = Toast.makeText(context, text, duration);
+                    toast.show();
+                    finish();
+                }
+            }
+        });
+    }
+
```
Comment (evanslaton) [inline app/src/main/java/com/hwang/xsighting/ViewSighting.java]: Does this empty line need to be here?

File: app/src/main/res/layout/activity_view_sighting.xml [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        android:layout_marginBottom="16dp"
+        android:text="Date"
+        android:textSize="18sp"
+        app:layout_constraintBottom_toTopOf="@+id/textView6"
+        app:layout_constraintEnd_toStartOf="@+id/postLocation"
+        app:layout_constraintStart_toEndOf="@+id/textView4"
+        app:layout_constraintTop_toBottomOf="@+id/postTtle" />
+
+    <TextView
+        android:id="@+id/postTtle"
```
Comment (evanslaton) [inline app/src/main/res/layout/activity_view_sighting.xml]: Mispelled postTitle

File: app/src/main/res/layout/activity_view_sighting.xml [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+        android:layout_marginEnd="8dp"
+        android:text="Sighting Title"
+        android:textAlignment="center"
+        android:textSize="24sp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/textView4"
```
Comment (evanslaton) [inline app/src/main/res/layout/activity_view_sighting.xml]: Should this have a more meaningful ID?

File: app/src/main/res/layout/activity_view_sighting.xml [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+        android:layout_marginBottom="16dp"
+        android:text="Location"
+        android:textSize="18sp"
+        app:layout_constraintBottom_toTopOf="@+id/textView6"
+        app:layout_constraintEnd_toStartOf="@+id/postUser"
+        app:layout_constraintStart_toEndOf="@+id/postDate"
+        app:layout_constraintTop_toBottomOf="@+id/postTtle" />
+
+    <TextView
+        android:id="@+id/textView6"
```
Comment (evanslaton) [inline app/src/main/res/layout/activity_view_sighting.xml]: Should this have a more meaningful ID?

File: app/src/main/java/com/hwang/xsighting/ViewSighting.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+                        toast.show();
+                        finish();
+                    }
+                } else {
+                    Log.d(TAG, "get failed with ", task.getException());
+                    //TODO: Redirect to the main page with a failure toast
+
+                    Context context = getApplicationContext();
+                    CharSequence text = "Something went wrong! Try Again";
+                    int duration = Toast.LENGTH_SHORT;
```
Comment (jeffborda) [inline app/src/main/java/com/hwang/xsighting/ViewSighting.java]: we can use: Toast.LENGTH_LONG if we want it to show up slightly longer

File: app/src/main/res/layout/activity_view_sighting.xml [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+        android:layout_marginBottom="8dp"
+        android:text="Description:"
+        android:textSize="18sp"
+        app:layout_constraintBottom_toTopOf="@+id/postDescription"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView4" />
+
+    <TextView
+        android:id="@+id/postUser"
```
Comment (evanslaton) [inline app/src/main/res/layout/activity_view_sighting.xml]: Is this the postUsername or postUserId?

File: app/src/main/res/layout/activity_view_sighting.xml (hunk 3 above)
Comment (evanslaton) [inline app/src/main/res/layout/activity_view_sighting.xml]: No blank line at the end

File: app/src/main/java/com/hwang/xsighting/ViewSighting.java [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+import com.hwang.xsighting.models.Sighting;
+
+public class ViewSighting extends AppCompatActivity {
+
+    //TODO: Check that as 'final' it does not cause issues
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private final String sightingId = getIntent().getStringExtra("SIGHTING_ID");
+    private final String TAG = "SightingDetail";
+    private Sighting sightingToDisplay;
+
```
Comment (evanslaton) [inline app/src/main/java/com/hwang/xsighting/ViewSighting.java]: Too many empty lines

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (JessLovell) [reply app/src/main/AndroidManifest.xml]: I fixed it

File: app/src/main/java/com/hwang/xsighting/ViewSighting.java (hunk 4 above)
Comment (JessLovell) [reply app/src/main/java/com/hwang/xsighting/ViewSighting.java]: ok

**Pull request merged**